src/main/java/org/folio/entlinks/domain/repository/AuthorityDataStatRepository.java

```diff
@@ -14,10 +13,10 @@ public interface AuthorityDataStatRepository extends JpaRepository<AuthorityData
 
   @Query(value = """
     select * from authority_data_stat
-    where action = :action and started_at between :startedAtStart and :startedAtEnd
+    where action = cast(:action as AuthorityDataStatAction) and started_at between :startedAtStart and :startedAtEnd
     order by started_at DESC
     limit :limit""", nativeQuery = true)
-  List<AuthorityDataStat> findByDateAndAction(@Param("action") AuthorityDataStatAction action,
+  List<AuthorityDataStat> findByDateAndAction(@Param("action") String action,
```

Comment:
Why this is needed? Hibernate should handle enum conversion as it is defined as @Type(PostgreSQLEnumType.class)

Reply:
Why do you use native query? It's pretty simple and could be in jpql.

```java
  @Query("""
    select a from AuthorityDataStat a
    where a.action = :action and a.startedAt between :startedAtStart and :startedAtEnd""")
  List<AuthorityDataStat> findByActionAndStartedAtBetween(@Param("action") AuthorityDataStatAction action,
                                                          @Param("startedAtStart") Timestamp startedAtStart,
                                                          @Param("startedAtEnd") Timestamp startedAtEnd,
                                                          Pageable pageable);
```

and use pageable for limit and sort 
```java
PageRequest.of(0, limit, Sort.by(Sort.Order.desc("startedAt")))
```